tsdb/head.go

```diff
@@ -2023,6 +2023,9 @@ type memSeries struct {
 	lset labels.Labels
 	meta *metadata.Metadata
 
+	// Series labels hash to use for sharding purposes.
+	shardHash uint64
```

Comment:
This line is the main place where people who don't need sharding will pay a price.
As Go buckets allocations into a smaller set of sizes it may make no practical difference.  Worth checking though.

Reply:
Let's keep this discussion in Bartek's comment: https://github.com/prometheus/prometheus/pull/10421#discussion_r1462931716